src/hypothesis/core.py

```diff
@@ -846,10 +864,26 @@ def run(self):
                 report(traceback.format_exc())
 
             if not raised_exception:
-                report(
-                    'Failed to reproduce exception. Expected: \n' +
-                    falsifying_example.__expected_exception,
-                )
+                if (
+                    isinstance(
+                        falsifying_example.__expected_exception,
+                        DeadlineExceeded
+                    ) and self.__test_runtime is not None
+                ):
+                    report((
+                        'Unreliable test timings! On an initial run, this '
```

Comment:
That won't necessarily help - if the *variation* is expected then this might happen at any deadline. I think if that's expected maybe the thing to do is turn off deadlines for this test.